src/main/java/org/testng/internal/TestInvoker.java

```diff
@@ -573,6 +573,7 @@ private ITestResult invokeMethod(
       runInvokedMethodListeners(BEFORE_INVOCATION, invokedMethod, testResult);
 
       if (arguments.getTestMethod() instanceof IInvocationStatus) {
+        ((IInvocationStatus) arguments.getTestMethod()).setInvokedAt(invokedMethod.getDate());
```

Comment:
Why don't you `.markAsInvoked()` here?

Reply:
Got rid of that and now just use `setInvokedAt()`